PYME/LMVis/visCore.py

```diff
@@ -792,13 +792,16 @@ def OpenFile(self, filename, recipe_callback=None):
         while len(self.layers) > 0:
             self.layers.pop()
         
-        print('Creating Pipeline')
+        logger.debug('Creating Pipeline')
         if filename is None and not ds is None:
             self.pipeline.OpenFile(ds=ds)
         else:
             args = self._populate_open_args(filename)
+            if args is None:
+                logger.info("OpenFile was canceled or not a valid file format")
```

Comment:
```suggestion
                #TODO - should we raise an exception in _populate_open_args() instead? Do we need a hard-fail here?
                logger.info("OpenFile was canceled or not a valid file format")
```

Reply:
I have added a commit to do all logging and checking within _populate_open_args() since it knows the details what failed and why. I would not generally raise an exception if a user canceled but have added a warning example that brings up a GUI msg for the user when in VisGUI. Please have a look and edit as required.

In other cases `_populate_open_args(`) can in future raise exception if warranted (for example, if we add new checks or file formats to this function).

`OpenFile` now just aborts if a None return value is received, assuming that all logging/warning is done already.